internal/builders/go/pkg/config.go

```diff
@@ -125,7 +135,24 @@ func validateMain(cf *goReleaserConfigFile) error {
 	}
 
 	// Validate the main path is under the current directory.
-	return pathIsUnderCurrentDirectory(*cf.Main)
+	if err := utils.PathIsUnderCurrentDirectory(*cf.Main); err != nil {
+		return convertPathError(err, "PathIsUnderCurrentDirectory")
+	}
+	return nil
+}
+
+func convertPathError(e error, msg string) error {
+	// TODO(https://github.com/slsa-framework/slsa-github-generator/issues/599): use same error contructions.
+	var errInternal *utils.ErrInternal
+	var errPath *utils.ErrInvalidPath
```

Comment:
nit: you can move these inside the if block below

Reply:
good catch. Done.